### PR TITLE
Revamp inproc tests (/FunctionsV2) for perf

### PR DESCRIPTION
**Pull request by andystaples**
# Summary
## What changed?
- Massive restructure of the inproc tests (/FunctionsV2 and /Common) to improve CI performance
  - Refactors the massive DurableTaskE2E tests into smaller E2E test classes containing focused groups of tests
  - Removes /Common as it is no longer needed, move all relevant tests to one common folder. Structure class files in test correctly
  - Revamps slow tests (reduce timeout durations, add polling where it is helpful)
  - Fixes the previously broken DurableOptionsConfigurationTests
  - Updates CI to run only the intended tests in each workflow using test traits

## Why is this change needed?
- Improve CI performance for quicker iteration

---

# Project checklist
- [x] Documentation changes are not required
  - [ ] Otherwise: Documentation PR is ready to merge and referenced in `pending_docs.md`
- [x] Release notes are not required for the next release
  - [ ] Otherwise: Notes added to `release_notes.md`
- [x] Backport is not required
  - [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
- [x] All required tests have been added/updated (unit tests, E2E tests)
- [x] No extra work is required to be leveraged by OutOfProc SDKs
  - [ ] Otherwise: Work tracked here: #issue_or_pr_in_each_sdk
- [x] No change to the version of the `WebJobs.Extensions.DurableTask` package
  - [ ] Otherwise: Major/minor updates are reflected in `/src/Worker.Extensions.DurableTask/AssemblyInfo.cs`
- [x] No EventIds were added to `EventSource` logs
  - [ ] Otherwise: Ensure EventIds are within the supported range in the existing Windows infrastructure (validate via deployed telemetry). If needed, extend the range via a PR such as https://msazure.visualstudio.com/One/_git/AAPT-Antares-Websites/pullrequest/7463263?_a=files
- [ ] This change should be added to the `v2.x` branch
  - [x] Otherwise: This change applies exclusively to `WebJobs.Extensions.DurableTask` v3.x and will be retained only in the `dev` and `main` branches
- [ ] Breaking change?
  - [ ] If yes:
    - Impact:
    - Migration guidance:
---

# AI-assisted code disclosure (required)
## Was an AI tool used? (select one)
- [ ] No
- [ ] Yes, AI helped write parts of this PR (e.g., GitHub Copilot)
- [x] Yes, an AI agent generated most of this PR

If AI was used:
- Tool(s): VS Code Copilot
- AI-assisted areas/files: Nearly all
- What you changed after AI output: Validate pipelines, specific test values, sanity checks

AI verification (required if AI was used):
- [x] I understand the code and can explain it
- [x] I verified referenced APIs/types exist and are correct
- [x] I reviewed edge cases/failure paths (timeouts, retries, cancellation, exceptions)
- [x] I reviewed concurrency/async behavior
- [x] I checked for unintended breaking or behavior changes

### CODE REVIEW

File: test/FunctionsV2/Helpers/SharedTraceSession.cs [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        /// when the last subscriber is removed.
+        /// </summary>
+        public static void Unsubscribe(int subscriberId)
+        {
+            Subscribers.TryRemove(subscriberId, out _);
+
+            lock (SyncLock)
+            {
+                refCount--;
+                if (refCount <= 0 && session != null)
```
Comment (bachuv) [inline test/FunctionsV2/Helpers/SharedTraceSession.cs]: Should we move the `TryRemove()` call into the lock?
```suggestion

            lock (SyncLock)
            {
                Subscribers.TryRemove(subscriberId, out _);
                refCount--;
                if (refCount <= 0 && session != null)
```

File: .github/workflows/validate-build.yml [hunk 2]
```diff
@@ -30,41 +29,68 @@ jobs:
...
+        for ($i = 1; $i -le $timeout; $i++) {
+          try {
+            $tcp = [System.Net.Sockets.TcpClient]::new('127.0.0.1', 10000)
+            $tcp.Dispose()
+            Write-Host "Azurite is ready"
+            break
+          } catch {
+            if ($i -eq $timeout) { throw "Azurite did not start within $timeout seconds" }
+            Write-Host "Waiting for Azurite ($i/$timeout)"
+            Start-Sleep -Seconds 1
```
[the remark this answers is not in the document]
Comment (andystaples) [reply .github/workflows/validate-build.yml]: No.

File: test/FunctionsV2/Helpers/SharedTraceSession.cs [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            Action<TraceEvent> callback,
+            IDictionary<string, TraceEventLevel> providers,
+            IDictionary<string, IEnumerable<int>> eventIdFilters = null)
+        {
+            int id = Interlocked.Increment(ref nextSubscriberId);
+            Subscribers[id] = callback;
+
+            lock (SyncLock)
+            {
+                refCount++;
```
Comment (bachuv) [inline test/FunctionsV2/Helpers/SharedTraceSession.cs]: Can we move `int id = Interlocked.Increment(ref nextSubscriberId);` and `Subscribers[id] = callback;` inside the lock? You can try replacing `Interlocked.Increment(ref nextSubscriberId)` with `++nextSubscriberId`.

Something like 
```suggestion
            lock (SyncLock)
            {
                int id = Interlocked.Increment(ref nextSubscriberId);
                Subscribers[id] = callback;
                refCount++;
```

Comment (andystaples) [reply test/FunctionsV2/Helpers/SharedTraceSession.cs]: Done

File: test/FunctionsV2/Helpers/SharedTraceSession.cs (hunk 1 above)
Comment (andystaples) [reply test/FunctionsV2/Helpers/SharedTraceSession.cs]: Done

**Review by sophiatev — APPROVED** [review]

**Review by sophiatev — APPROVED** [review]

**Review by sophiatev — APPROVED** [review]

**Review by bachuv — APPROVED** [review]
